### DISCUSSION TITLE
Suggestion: Add zoom steps

### DISCUSSION THREAD

**Discussion by giwul70** [sumatrapdfreader/sumatrapdf#4431] [Ideas]
I shortly check previous threads, probably I am overlooking them but I could not find anything on 'zoom steps'.
If the suggestion has been made before, then please do accept my apologies.

If not, what about adding an option to set 'user defined' zoom steps (for instance 10%, 15% or 20%)  and that be visible in the toolbar?
At every click on the zoom button it increases by (e.g.) 10%
Or one can click in the "percentage box" (in the below example 100%), do ctrl arrow up/down or enter e.g. 150

Positively meant suggestion.

Keep up the good work!

I meant something like the below.

![image](https://github.com/user-attachments/assets/d775a026-b55d-4b25-a62f-cc49f85778d4)

**Comment by GitHubRulesOK** [comment]
the steps can be use 2 ways one as percent's (zoom Increments) is cumulative so not often used thus if at 50% and zoom +10% it would be I think 55% then 61.5% then 67.2% (long time since testing so you would need to experiment) Thus most users do not use that as "odd"

The other "normal" method is listed numbered steps
```
ZoomLevels = 8.33 12.5 18 25 33.33 50 66.67 75 100 125 150 200 300 400 600 800 1000 1200 1600 2000 2400 3200 4800 6400
```
where you can ADD `150` but it is already there so will be listed in CTRL+Y and can be assigned to its own "hotkey"

![image](https://github.com/user-attachments/assets/bb2028db-0c30-45b8-b381-ca1d20f5f796)

**Comment by giwul70** [comment]
Thank you. I am not an expert, I use this splendid reader 'as it is'. There is not much I can change, maybe even nothing... :-)
For a moment I thought I could change the zoomlevels by changing the above valuess (in SumatraPDF-settings.txt). That did not work, the zoom levels are hard coded I suppose.
Maybe one day smaller steps between 50 and -say- 150 or 200 will be implemented.
Anyway, thanks again.

**Reply by GitHubRulesOK** [reply]
The edits will WORK but not be SEEN unless you use query custom current value so hard code 75 is missing

```
ZoomLevels = 8.33 12.5 18 25 33.33 50 60 70 80 90 100 125 150 200 300 400 600 800 1000 1200 1600 2000 2400 3200 4800 6400
```
![image](https://github.com/user-attachments/assets/74241853-3b4d-4d9f-987a-7fb215723f8a)

![image](https://github.com/user-attachments/assets/f6c2c4ac-27ba-411e-b394-a497a7bc76d4)

**Reply by GitHubRulesOK** [reply]
@giwul70  you can set keys in latest pre-release like this

```
Shortcuts [
    [
        Cmd = CmdZoomCustom 60%
        Key = 6
    ]
    [
        Cmd = CmdZoomCustom 70%
        Key = 7
    ]
]
```
press 7 will zoom like this
![image](https://github.com/user-attachments/assets/c6cc60b9-5178-4036-b32a-f38108600ce9)

**Reply by GitHubRulesOK** [reply]
@kjk 
the timout for the zoom message (even on this slow device :-) is too quick can it be a second or 2 longer ?

**Reply by kjk** [reply]
@GitHubRulesOK increased from 300 ms to 2 secs

**Reply by giwul70** [reply]
@GitHubRulesOK

Thank you!

I tried this (edit the SumatraPDF-settings.txt file)

Shortcuts [
	[
		Cmd = CmdZoomCustom 50%
		Key = 5
	]
	[
		Cmd = CmdZoomCustom 60%
		Key = 6
	]
	[
		Cmd = CmdZoomCustom 70%
		Key = 7
	]
	[
		Cmd = CmdZoomCustom 80%
		Key = 8
	]
	`[
		Cmd = CmdZoomCustom 90%
		Key = 8
	]
]



![SnagIt-04082024 102634](https://github.com/user-attachments/assets/88e02e56-b648-44e3-99f0-9d8668fca38c)

regretfully it did not work.
I am sure I must be doing something wrong.
(3.6.16616)

**Reply by GitHubRulesOK** [reply]
Hmm looks ok
apart from your editor is not the normal basic Notepad where those Tabs and EOLs would not be visible in "Notepad UTF-8 with BOM" editor. 
So there is ONE BIG issue with 32bits that the edits must be done without SumatraPDF "active" or else may be flushed on closure !!
If I add those entries in a session and save then close SumatraPDF X32 those edits will not be there in next session !

**Comment by giwul70** [comment]
Thanks. Solved.
FWIW: Am using EditPadPro v8, the file was UTF8 with BOM, changed it to UTF8 No Bom.
Made no difference.
I made the changes outside SumatraPDF.
This time I did it from within (noticed my previous settings, i.e. made whilst SumatraPDF was -ot- running) weren't there.
So I added the above whilst SumatraPDF was running now. Didn't know I should update it from within SumataPDF.

Anyway, it worked out fine.

Many thanks again!

![SnagIt-04082024 131630](https://github.com/user-attachments/assets/09d29c18-f84c-4222-a1f6-d26880c8e277)

**Reply by GitHubRulesOK** [reply]
64 bit edits will be preserved if saved whilst running 
but 32 bits had an issue that was resolved by needing edits without running so there is a significant difference there !

**Comment by giwul70** [comment]
Am running SumatraPDF x64 on (good old) Windows 10 x64, so things went fine then.

**Comment by kjk** [comment]
@giwul70 The issue is as follows:

In 64-bit you don't have to restart SumatraPDF for new settings to take effect. We auto-detect changes to settings file, re-load the file and apply the changes.

In 32-bit you have to exit and start SumatraPDF again for settings changes to take effect.

**Reply by giwul70** [reply]
Thanks.
SumatraPDF is really super. It is my default PDF reader. Only a few specific tasks I perform with Acrobat (combining image files, OCR, moving pages from one PDF to the other, scanning documents from printer ADF, basically all having to do with 'creating/editing' PDF files)

**Comment by GitHubRulesOK** [comment]
@kjk 
its not as clear cut for 32 bit as your comment
If I add an advanced setting in 32 bit it is naturally going to ask me to save the setting and notepad diligently saves the txt
![image](https://github.com/user-attachments/assets/5a853e8e-471f-4d6b-834f-9c531abb1d66)
But on close session the file edit is discarded and on reopen it is not there 
![image](https://github.com/user-attachments/assets/bacf4a28-964e-4a0e-a197-edb20fc41040)


The session MUST be closed to make any edits. but it is not easy to make an edit as when session is open.
This means you keep settings open and close session but that has the opposite effect of discarding session changes !!
